kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/ProducerStateManager.java

```diff
@@ -182,8 +187,10 @@ public void updateMapEndOffset(long mapEndOffset) {
 
     public void updateTxnIndex(CompletedTxn completedTxn, long lastStableOffset) {
         if (completedTxn.isAborted()) {
-            abortedIndexList.add(new AbortedTxn(completedTxn.producerId(), completedTxn.firstOffset(),
-                    completedTxn.lastOffset(), lastStableOffset));
+            synchronized (abortedIndexList) {
```

Comment:
Great change! I noticed yesterday that the updates to `abortedIndexList` weren't thread safe. My only question is whether we should just use a copy on write array list? I would expect we'll read the array much more often than we'll write to it, though that is use case dependent.